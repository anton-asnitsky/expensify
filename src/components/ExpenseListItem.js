import React from 'react';
import { connect } from 'react-redux';
import { removeExpense } from '../actions/expenses';

const ExpenseListItem = ({dispatch, id, description, amount, createdAt}) => {
    return (
        <li>
            <p>{description}</p>
            <p>{amount} - {createdAt}</p>
            <button
                onClick={(e) => {
                    dispatch(removeExpense({id}));
                }}
            >Remove</button>
        </li>
    );
};

export default connect()(ExpenseListItem);