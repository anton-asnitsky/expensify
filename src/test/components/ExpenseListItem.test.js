import React            from 'react';
import ExpenseListItem  from '../../components/ExpenseListItem';
import { shallow }      from 'enzyme';
import expeneses         from '../fixtures/expenses';

test('should render ExpneseListItem with expnese data', () => {
    const wrapper = shallow(<ExpenseListItem key={expeneses[0].id} {...expeneses[0]} />);

    expect(wrapper).toMatchSnapshot();
});