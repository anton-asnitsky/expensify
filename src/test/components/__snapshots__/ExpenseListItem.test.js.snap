// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpneseListItem with expnese data 1`] = `
<li>
  <Link
    replace={false}
    to="/edit/1"
  >
    <h3>
      Gum
    </h3>
  </Link>
  <p>
    $1.95
    -
    January 1st 1970
  </p>
</li>
`;
